### PR TITLE
feature to create a lookup-table to raytrace sensor fovs

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: sensor_placement/common/src/particle.cpp [hunk 1]
```diff
@@ -537,6 +558,91 @@ void particle::updateTargetsInfo(size_t sensor_index)
...
+  unsigned int rays_checked = 0;
+  unsigned int ray = ray_start;
+
+  while(rays_checked < number_of_rays_to_check)
+  {
+    for(unsigned int cell=0; cell < sensors_.at(sensor_index).getLookupTable().at(ray).size(); cell++)
+    {
+      //absolute x and y coordinates of the current cell
+      int x = sensor_pose.position.x + sensors_.at(sensor_index).getLookupTable().at(ray).at(cell).x;        
+      int y = sensor_pose.position.y + sensors_.at(sensor_index).getLookupTable().at(ray).at(cell).y;
```
Comment (fmirus) [inline sensor_placement/common/src/particle.cpp]: is this correct? should x and y be cell indices or world coordinates?

**Comment by fmirus** [comment]
This version of the raytracing seems to work fine as a first draft. I suggest we merge this Pull Request into groovy_dev and start new features for Visualization resp. Optimiztation. Is lookup_table.cpp still needed? If not, it should be removed before the merge as well as all "new"-comments. Any other ideas? Discussion welcome :-)

**Comment by ghost** [comment]
The current version only works with rectangular polygons, because the targets_with_info are stored in a 1-dimensional vector.

**Comment by fmirus** [comment]
After discussion the last comment seems not to be a problem or bug, but ipa-mig-fb will check again to make sure.

**Comment by fmirus** [comment]
manually merged into groovy_dev

**Pull request merged**